### PR TITLE
OVMM: Support for integrated baseline

### PR DESCRIPTION
**Pull request by yvsriram**
## Motivation and Context

- Support for using home-robot wrapper env
- Configs for end-to-end task
- Update arm action space to support discrete joint action
- Support evaluation on selected episodes
- Add Receptacle segmentation and camera pose sensors
- New success measures for measuring partial success

## How Has This Been Tested

By evaluating baseline in home-robot with minival episode set 

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py [hunk 1]
```diff
@@ -135,18 +128,68 @@ def reset(self, episode: Episode):
...
+            episode, start_hold_obj_idx=start_hold_obj_idx, force_idx=self.force_obj_to_idx
+        )
+
         self._nav_to_info = self._generate_nav_start_goal(
-            episode, force_idx=self.force_obj_to_idx
+            episode, nav_to_pos, start_hold_obj_idx=start_hold_obj_idx
         )
-        sim.robot.base_pos = self._nav_to_info.robot_start_pos
-        sim.robot.base_rot = self._nav_to_info.robot_start_angle
+        if self._pick_init:
```
Comment (JHurricane96) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:156]: What does `_pick_init` signify?

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py [hunk 2]
```diff
@@ -54,6 +54,8 @@ def __init__(self, *args, config, dataset=None, **kwargs):
         self.robot_start_rotation = None
 
         self._min_start_distance = self._config.min_start_distance
+        self._goal_type = config.goal_type
```
Comment (JHurricane96) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:57]: What does `goal_type` mean and what values can it take?

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py (hunk 1 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:164]: Added comments, testing a change where this is moved outside this file

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py (hunk 2 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:57]: this is supposed to distinguish ovmm end-to-end task from the other nav skills. Testing a change where end-to-end task uses a separate ovmm_task

File: habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      - robot_third_rgb
+      - start_receptacle
+      - goal_receptacle
+      - robot_start_gps
+      - robot_start_compass
+      - joint
+      - object_segmentation
+      - start_recep_segmentation
+      - goal_recep_segmentation
+      - object_category
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml]: yes

File: habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      - data/objects/ycb/configs/
+      - data/objects/amazon_berkeley/configs/
+      - data/objects/google_object_dataset/configs/
+    debug_render_goal: False
+    concur_render: True
+    auto_sleep: True
+    kinematic_mode: True
+    agents:
+      main_agent:
+        radius: 0.25
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml]: Changed to 0.3

File: habitat-lab/habitat/datasets/rearrange/rearrange_dataset.py [hunk 5]
```diff
@@ -191,6 +197,11 @@ def to_json(self) -> str:
     @staticmethod
     def __deserialize_goal(serialized_goal: Dict[str, Any]) -> ObjectGoal:
         g = ObjectGoal(**serialized_goal)
+        # if the view points are not cached separately, read from original episodes
+        for vidx, view in enumerate(g.view_points):
+            view_location = ObjectViewLocation(**view)  # type: ignore
+            view_location.agent_state = AgentState(**view_location.agent_state)  # type: ignore
+            g.view_points[vidx] = view_location
```
Comment (JHurricane96) [inline habitat-lab/habitat/datasets/rearrange/rearrange_dataset.py:204]: This causes an error when viewpoints are cached separately. Add a condition here.

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py (hunk 2 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:57]: Let me see if I can get this done before the PR is merged. Otherwise, will handle this later as discussed.

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 6]
```diff
@@ -148,11 +148,13 @@ def __init__(
     ):
         super().__init__()
         self.no_downscaling = no_downscaling
+        # Goal/privileged information sensors
```
Comment (cpaxton) [inline habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:151]: should we add an explanation?
also it seems weird that this is hard-coded?

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 7]
```diff
@@ -406,6 +422,21 @@ class ObjectSegmentationSensorConfig(LabSensorConfig):
     dimensionality: int = 256
 
 
+@attr.s(auto_attribs=True, slots=True)
+class StartRecepSegmentationSensorConfig(ObjectSegmentationSensorConfig):
```
Comment (cpaxton) [inline habitat-lab/habitat/config/default_structured_configs.py:426]: i don't really understand what we're doing here

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 8]
```diff
@@ -621,6 +653,63 @@ class NavToObjSuccessMeasurementConfig(MeasurementConfig):
...
+    type: str = "NavOrientToPlaceSucc"
+
+
+@attr.s(auto_attribs=True, slots=True)
+class PickNavToPlaceSuccMeasurementConfig(MeasurementConfig):
+    type: str = "PickNavToPlaceSucc"
+
+
+@attr.s(auto_attribs=True, slots=True)
+class PickNavOrientToPlaceSuccMeasurementConfig(MeasurementConfig):
```
Comment (cpaxton) [inline habitat-lab/habitat/config/default_structured_configs.py:709]: Same with all these

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 9]
```diff
@@ -827,7 +911,7 @@ class TaskConfig(HabitatBaseConfig):
     # Spawn parameters
     physics_stability_steps: int = 1
     num_spawn_attempts: int = 200
-    spawn_max_dists_to_obj: float = 2.0
+    spawn_max_dists_to_obj: float = 1.0
```
Comment (cpaxton) [inline habitat-lab/habitat/config/default_structured_configs.py:914]: is this from object or from viewpoint?

File: habitat-lab/habitat/tasks/rearrange/__init__.py [hunk 10]
```diff
@@ -51,3 +52,9 @@ def _try_register_rearrange_task():
         HabitatSimActions.extend_action_space("empty")
     if not HabitatSimActions.has_action("rearrange_stop"):
         HabitatSimActions.extend_action_space("rearrange_stop")
+    if not HabitatSimActions.has_action("extend_arm"):
+        HabitatSimActions.extend_action_space("extend_arm")
+    if not HabitatSimActions.has_action("face_arm"):
+        HabitatSimActions.extend_action_space("face_arm")
+    if not HabitatSimActions.has_action("reset_joints"):
+        HabitatSimActions.extend_action_space("reset_joints")
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/__init__.py:60]: are all of these implemented on the robot side? are any missing?

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 11]
```diff
@@ -125,6 +125,56 @@ def step(self, is_last_action, *args, **kwargs):
...
+class FaceArmAction(RobotAction):
+    """
+        The camera joints are set such that the camera points towards the arm. Used when the agent switches from navigation mode to manipulation mode
+    """
+    def step(self, *args, is_last_action, **kwargs):
+        face = kwargs.get("face_arm", [-1.0])
+        if face[0] > 0:
+            motor_pos = self._sim.robot.arm_motor_pos.tolist()
+            joint_pos = self._sim.robot.arm_motor_pos.tolist()
+            motor_pos[8] = -1.7375
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:153]: What is this value?

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 12]
```diff
@@ -125,6 +125,56 @@ def step(self, is_last_action, *args, **kwargs):
...
+
+@registry.register_task_action
+class ResetJointsAction(RobotAction):
+    """
+        The arm and camera joints are reset to resting position. The camera faces the base and is tilted downwards. The arm is retracted with the gripper facing downwards. Used for switching to navigation mode.
+    """
+    def step(self, *args, is_last_action, **kwargs):
+        reset = kwargs.get("reset_joints", [-1.0])
+        if reset[0] > 0:
+            self._sim.robot.arm_motor_pos = [0, 0, 0, 0, 0.775, 0, -1.57000005, 0, 0.0, -0.7125]
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:170]: for magic numbers like these, can we load them from home-robot? or pass them in somehow?

File: habitat-lab/habitat/tasks/rearrange/ovmm_sensors.py [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+                [
+                    view_point.agent_state.position
+                    for goal in episode.candidate_objects
+                    for view_point in goal.view_points
+                ],
+                axis=0,
+        )
+
+@registry.register_measure
+class RotDistToPickGoal(CatNavRotDistToGoal, Measure):
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/ovmm_sensors.py:42]: Do we need descriptions for these?

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py [hunk 14]
```diff
@@ -135,18 +140,116 @@ def reset(self, episode: Episode):
 
         # in the case of Stretch, force the agent to look down and retract arm with the gripper pointing downwards
         if isinstance(sim.robot, StretchRobot):
+            camera_rot = (
+                -1.7375 if self._pick_init or self._place_init else 0.0
+            )
             sim.robot.arm_motor_pos = np.array(
-                [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, 0.0, -0.7125]
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:139]: again i really dislike having all these hard coded values - espeically if they are intended to be same as real world. we should put them elsewhere

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py [hunk 15]
```diff
@@ -135,18 +140,116 @@ def reset(self, episode: Episode):
...
+            # in the case of Stretch, rotate base so that the arm faces the target location
+            if isinstance(self._sim.robot, StretchRobot):
+                sim.robot.base_rot = angle_to_obj + np.pi / 2
+            else:
+                sim.robot.base_rot = angle_to_obj
+        elif self._place_init:
+            # spawn agent close to and oriented towards goal receptacle for place testing, [wip]: moving to ovmm_task.py
+            if isinstance(sim.robot, StretchRobot):
+                sim.robot.arm_motor_pos = np.array(
+                    [0.0] * 4
```
Comment (cpaxton) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:211]: hard coded values

**Review by cpaxton — COMMENTED** [review]
mostly looks good, just a couple minor comments

File: habitat-baselines/habitat_baselines/common/construct_vector_env.py [hunk 16]
```diff
@@ -18,6 +18,7 @@ def construct_envs(
     config: "DictConfig",
     workers_ignore_signals: bool = False,
     enforce_scenes_greater_eq_environments: bool = False,
+    make_env_fn = make_gym_from_config
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply habitat-baselines/habitat_baselines/common/construct_vector_env.py:21]: This is just the default parameter. On `home-robot` side, we pass in a custom wrapper. Here, this is hardcoded to `make_gym_from_config`

File: habitat-baselines/habitat_baselines/common/construct_vector_env.py [hunk 17]
```diff
@@ -93,7 +94,7 @@ def construct_envs(
         vector_env_cls = VectorEnv
 
     envs = vector_env_cls(
-        make_env_fn=make_gym_from_config,
+        make_env_fn=make_env_fn,
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply habitat-baselines/habitat_baselines/common/construct_vector_env.py:97]: Same as above comment. This will allow us to pass custom env wrapper [here](https://github.com/facebookresearch/home-robot/blob/4141a5f7093246a1861d12f31b621e6b8d755066/projects/habitat_ovmm/eval_vectorized.py#L72)

File: habitat-lab/habitat/tasks/rearrange/__init__.py (hunk 10 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/__init__.py]: Yeah these were all implemented. We were just using different names (eg. manipulation_mode, navigation_mode). I have removed these now though. Only arm_action (for joints) and base control action are exposed

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py (hunk 11 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/actions/actions.py]: Removed this from here

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py (hunk 12 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/actions/actions.py]: Moved to home-robot

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 18]
```diff
@@ -148,11 +148,13 @@ def __init__(
...
+        skip_sensors = {ImageGoalSensor.cls_uuid, "robot_head_panoptic"}
```
Comment (ykarmesh) [inline habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:152]: Would really prefer to use the UUID here

File: habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml [hunk 19]
```diff
@@ -0,0 +1,76 @@
+# @package _global_
+
+defaults:
+  - /habitat: habitat_config_base
+  - /habitat/simulator/agents@habitat.simulator.agents.main_agent: rgbdp_head_rgb_third_agent
+  - /habitat/task/rearrange: ovmm
+  - /habitat/dataset/rearrangement: floorplanner
```
Comment (ykarmesh) [inline habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml:7]: Are we changing the name to hssd or pushing that to a future pr?

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/pick_task.py [hunk 20]
```diff
@@ -13,6 +13,7 @@
 from habitat.robots.stretch_robot import StretchRobot
 from habitat.tasks.rearrange.rearrange_task import RearrangeTask
 from habitat.tasks.rearrange.utils import get_robot_spawns, rearrange_logger
+import habitat.robots.stretch_robot as robot_class
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/pick_task.py:16]: This import wasnt used?

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 7 above)
Comment (ykarmesh) [reply habitat-lab/habitat/config/default_structured_configs.py:429]: This is some weird hydra stuff that Habitat does. I also dont understand the reason for this extra step

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 21]
```diff
@@ -624,6 +656,63 @@ class NavToObjSuccessMeasurementConfig(MeasurementConfig):
...
+    type: str = "DistToPickGoal"
+
+
+@attr.s(auto_attribs=True, slots=True)
+class DistToPlaceGoalMeasurementConfig(DistToGoalMeasurementConfig):
+    type: str = "DistToPlaceGoal"
+
+
+@attr.s(auto_attribs=True, slots=True)
+class CatNavRotDistToGoalMeasurementConfig(RotDistToGoalMeasurementConfig):
```
Comment (ykarmesh) [inline habitat-lab/habitat/config/default_structured_configs.py:670]: You have only added `Cat` in this. Is that a mistake or there is some reason?

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 9 above)
Comment (yvsriram) [reply habitat-lab/habitat/config/default_structured_configs.py:917]: It is still obj in this PR. Will change this naming in the upcoming PR which spawns wrt viewpoints

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py (hunk 18 above)
Comment (yvsriram) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:152]: The uuid is in [default_structured_configs](https://github.com/facebookresearch/habitat-lab/blob/df89d7c3c951ae02aeb0bb57c6ac975bb789247c/habitat-lab/habitat/config/default_structured_configs.py#L1064). Importing that here does not seem right. 

But @JHurricane96, is excluding this from gym.obs_keys not sufficient?

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 21 above)
Comment (yvsriram) [reply habitat-lab/habitat/config/default_structured_configs.py:670]: Will use the prefix 'Ovmm' for all these sensors

Comment (yvsriram) [reply habitat-lab/habitat/config/default_structured_configs.py:670]: No reason. I think just different people adding stuff for different things. CatNav was added for skill-wise nav and others were added for integrated metrics

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py [hunk 22]
```diff
@@ -82,7 +82,7 @@ def get_observation(
         pan_obs = observations["robot_head_panoptic"]
         max_obs_val = np.max(pan_obs)
         obs = np.zeros(
-            (self._dimensionality, self._dimensionality, self._num_channels),
+            (pan_obs.shape[0], pan_obs.shape[1], self._num_channels),
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py:85]: I think the observation space will have to change accordingly

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 21 above)
Comment (yvsriram) [reply habitat-lab/habitat/config/default_structured_configs.py]: Updated

File: habitat-lab/habitat/tasks/rearrange/ovmm_sensors.py (hunk 13 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/ovmm_sensors.py]: Added descriptions

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py (hunk 22 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py:85]: Yes I am fixing this properly for all sensors

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 8 above)
Comment (yvsriram) [reply habitat-lab/habitat/config/default_structured_configs.py]: Same as above. The are only for defining config params and the default values they can take

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py (hunk 22 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py]: Done.

File: habitat-lab/habitat/config/habitat/task/rearrange/ovmm.yaml [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+actions:
+  arm_action:
+    type: "ArmAction"
+    arm_controller: "ArmRelPosReducedActionStretch"
+    grip_controller: "GazeGraspAction"
+    arm_joint_mask: [1,0,0,0,1,1,1,1,1,1]
+    arm_joint_dimensionality: 7
+    grasp_thresh_dist: 0.15
+    disable_grip: False
+    delta_pos_limit: 6.28 # The range of joints are:  [ 0., 0.,  0.,  0.,  0., -1.75, -1.57, -3.14, -3.9, -1.53] -> [0.13, 0.13, 0.13, 0.13, 1.1 , 4., 0.56, 3.14, 1.5 , 0.79], so maximum delta is 6.28
```
Comment (yvsriram) [inline habitat-lab/habitat/config/habitat/task/rearrange/ovmm.yaml:67]: I got these values using `cur_robot.arm_joint_limits` in sim. @cpaxton, Jimmy asked me to confirm if these ranges match  real stretch. eg. if the real agent can realize joint_wrist_roll in the entire range [-3.14, 3.14]

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py (hunk 14 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py:139]: Since, these are used for as initial poses at the time of training skills this can't be pulled from home-robot. I moved these to a separate file in habitat-lab.

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py (hunk 15 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py]: Moved to a separate file in habitat-lab

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py (hunk 6 above)
Comment (yvsriram) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py]: Added an explanation. @JHurricane96 why was this necessary?

File: habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml (hunk 19 above)
Comment (yvsriram) [reply habitat-lab/habitat/config/benchmark/rearrange/ovmm.yaml]: Update this particular thing to hssd. There are still other places that need updating. Will do those later

File: habitat-lab/habitat/config/benchmark/rearrange/cat_gaze.yaml [hunk 24]
```diff
@@ -4,7 +4,7 @@ defaults:
   - /habitat: habitat_config_base
   - /habitat/simulator/agents@habitat.simulator.agents.main_agent: depthp_head_rgb_third_agent
   - /habitat/task/rearrange: cat_gaze
-  - /habitat/dataset/rearrangement: floorplanner
+  - /habitat/dataset/rearrangement: hssd
```
Comment (JHurricane96) [inline habitat-lab/habitat/config/benchmark/rearrange/cat_gaze.yaml:7]: Where is the new hssd dataset file?

Comment (yvsriram) [reply habitat-lab/habitat/config/benchmark/rearrange/cat_gaze.yaml:7]: Pushed

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py (hunk 18 above)
Comment (JHurricane96) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:152]: It looks like it is sufficient. We can remove this from here.

Comment (yvsriram) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:152]: Thanks for checking. Will remove

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py (hunk 6 above)
Comment (yvsriram) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py]: Removed now. This is not needed.

File: habitat-lab/habitat/config/habitat/task/rearrange/ovmm.yaml (hunk 23 above)
Comment (cpaxton) [reply habitat-lab/habitat/config/habitat/task/rearrange/ovmm.yaml:67]: I think this is correct.

**Review by cpaxton — APPROVED** [review]
lgtm

File: habitat-lab/habitat/config/habitat/dataset/rearrangement/hssd.yaml [hunk 25]
```diff
@@ -0,0 +1,12 @@
...
+defaults:
+  - /habitat/dataset: object_rearrange_dataset_config_schema
+  - _self_
+
+type: ObjectRearrangeDataset-v0
+split: train
+data_path: data/episodes/rearrange/v3/{split}/cat_npz-exp.json.gz
+scenes_dir: "data/fphab/fphab/"
+viewpoints_matrix_path: "data/episodes/rearrange/v3/{split}/cat_npz-exp-vps.npy"
+transformations_matrix_path: data/episodes/rearrange/v3/{split}/cat_npz-exp-trans.npy
```
Comment (JHurricane96) [inline habitat-lab/habitat/config/habitat/dataset/rearrangement/hssd.yaml:12]: newline

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 26]
```diff
@@ -148,11 +148,12 @@ def __init__(
     ):
         super().__init__()
         self.no_downscaling = no_downscaling
+        skip_sensors = {ImageGoalSensor.cls_uuid}
```
Comment (JHurricane96) [inline habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:151]: Can we revert this change completely? We don't need a separate set anymore.

**Review by JHurricane96 — APPROVED** [review]
